pymeasure/instruments/keysight/keysight34980A.py

```diff
@@ -0,0 +1,628 @@
+        """    
+        if (chan == ""):
+            self.write(":SENSe:RESistance:RANGe:AUTO OFF")
+            while((self.is_resistance_autorange()) == 1):
+                self.write(":SENSe:RESistance:RANGe:AUTO OFF")
+        else:
+            self.write(":SENSe:RESistance:RANGe:AUTO OFF (@" + str(chan) + ")")
+            while((self.is_resistance_autorange(chan)) == 1):
+                self.write(":SENSe:RESistance:RANGe:AUTO OFF (@" + str(chan) + ")")
+
+    def is_resistance_autorange(self, chan = ""):    
+        """
+        This command checks if autoranging for resistance measurements on the specified channel are enabled.       
+        """
+        if (chan == ""):
+            temp_values = self.values(":SENSe:RESistance:RANGe:AUTO?") 
+            return(str(temp_values[0]))
+        else:
+            temp_values = self.values(":SENSe:RESistance:RANGe:AUTO? (@" + str(chan) + ")") 
+            return(str(temp_values[0]))
+
+    def measure_dc_voltage(self, chan, resolution = "DEFault", range_param = "AUTO"):    
+        """
+        The following command configures the instrument for dc voltage measurements, 
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:VOLTage:DC?")
+        dc_voltage = str(temp_values[0])
+        return(dc_voltage)
+
+    def measure_ac_voltage(self, chan, resolution = "DEFault", range_param = "AUTO"):    
+        """
+        The following command configures the instrument for ac voltage measurements, 
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:VOLTage:AC?")
+        return(str(temp_values[0]))
+
+    def measure_dc_current(self, chan, resolution = "DEFault", range_param = "AUTO"):    
+        """
+        The following command configures the instrument for dc current measurements, 
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:CURRent:DC?")
+        return(str(temp_values[0]))
+
+    def measure_ac_current(self, chan, resolution = "DEFault", range_param = "AUTO"):
+        """
+        The following command configures the instrument for ac voltage measurements, 
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:CURRent:AC?")
+        return(str(temp_values[0]))
+
+    def measure_frequency(self, chan, resolution = "DEFault", range_param = "DEFault"):    
+        """
+        The following command configures the instrument for frequency measurements, 
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:FREQuency?")
+        return(str(temp_values[0]))
+
+    def measure_period(self, chan, resolution = "DEFault", range_param = "DEFault"):    
```

Comment:
The arguments (chan, resolution and range_param) can be removed, right? they're not used anyway